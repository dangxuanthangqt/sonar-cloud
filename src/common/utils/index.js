/* eslint-disable no-bitwise */
/* eslint-disable func-names */
export const handleRegexWithSpecialCharacters = (string) =>
  string.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1');

export const formatHint = (string) => string.replace(/\n/gm, '<br/><br/>');

export const getDateFormatString = () => {
  const locale = window.navigator.userLanguage || window.navigator.language;
  const formatObj = new Intl.DateTimeFormat(locale).formatToParts(new Date());

  return formatObj
    .map((obj) => {
      switch (obj.type) {
        case 'day':
          return 'dd';
        case 'month':
          return 'MM';
        case 'year':
          return 'yyyy';
        default:
          return obj.value;
      }
    })
    .join('');
};

export const truncateString = (str = '', num = 100) => {
  if (str.length > num) {
    return `${str.slice(0, num)} ...`;
  }

  return str;
};

export function uuidv4() {
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
    const r = (Math.random() * 16) | 0;
    const v = c === 'x' ? r : (r & 0x3) | 0x8;
    return v.toString(16);
  });
}
