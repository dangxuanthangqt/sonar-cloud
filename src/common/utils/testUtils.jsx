import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { StyledEngineProvider } from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';
import { muiTheme } from 'layouts/baseStyle';
import { QueryClient, QueryClientProvider } from 'react-query';
import { RecoilRoot } from 'recoil';
import { render as rtlRender } from '@testing-library/react';

function AllTheProviders({ children }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
      },
    },
  });

  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={muiTheme}>
        <QueryClientProvider client={queryClient}>
          <RecoilRoot>
            <Router>{children}</Router>
          </RecoilRoot>
        </QueryClientProvider>
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

const render = (ui, options) => {
  return rtlRender(ui, { wrapper: AllTheProviders, ...options });
};

export { render };
export * from '@testing-library/react';
