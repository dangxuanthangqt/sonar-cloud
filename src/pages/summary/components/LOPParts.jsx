/* eslint-disable react/no-array-index-key */
/* eslint-disable array-callback-return */
import { Box, Divider, Grid, Typography } from '@mui/material';
import { makeStyles } from '@mui/styles';
import { useDataSourceSummary } from 'hooks/use-data-source-summary';
import { useRecoilState } from 'recoil';
import React, { useMemo, useEffect } from 'react';
import DataSourceSummary from '@/components/data-source-summary';
import { lopsAndPartsStateAtom } from '@/recoil/atom/lops-and-parts-state';

const useCriteriaStyles = makeStyles((theme) => ({
  header: {
    backgroundColor: theme.palette.primary.light,
    paddingTop: 14,
    paddingBottom: 12,
    paddingLeft: theme.spacing(2),
    marginBottom: theme.spacing(1),
    '& .MuiTypography-root': {
      color: theme.palette.primary.main,
      fontSize: 16,
      fontWeight: 700,
    },
  },
  label: {
    color: theme.palette.secondary.main,
    fontSize: 18,
    fontWeight: 700,
    marginBlock: 12,
  },
  additional_label: {
    color: theme.colors.gray,
    fontSize: 16,
    fontWeight: 400,
  },
  data_source: {
    padding: 10,
    backgroundColor: theme.palette.primary.light,
    fontSize: 11,
    borderRadius: 4,
    fontWeight: 400,
    marginBlock: 2,
  },
  heading: {
    color: theme.palette.secondary.main,
    fontSize: 18,
    fontWeight: 700,
    marginBlock: 12,
  },
  data_source_1: {
    padding: 10,
    backgroundColor: theme.palette.secondary.tab_bg,
    fontSize: 11,
    borderRadius: 4,
    fontWeight: 400,
    marginBlock: 2,
  },
  keyword: {
    backgroundColor: theme.palette.secondary.ligt_bg,
    borderRadius: 6,
  },
  keyword_bolts: {
    fontSize: 13,
    marginLeft: 5,
    fontWeight: 600,
  },
  keyword_heading: {
    backgroundColor: theme.palette.primary.light,
    padding: 9,
    fontSize: 11,
    display: 'flex',
    alignContent: 'center',
    alignItems: 'center',
    borderRadius: 4,
    fontWeight: 400,
  },
  keyword_applied: {
    color: theme.colors.gray,
    fontSize: 14,
    fontWeight: 400,
  },
  inner_box: {
    paddingLeft: 18,
  },
  load_in: {
    color: theme.colors.gray,
    fontSize: 16,
    fontWeight: 400,
    marginLeft: 18,
    marginTop: 7,
  },
  date_value_label: {
    color: theme.palette.secondary.main,
    fontSize: 14,
    paddingLeft: 6,
    fontWeight: 500,
    padding: 8,
    marginLeft: 15,
    lineHeight: '200%',
    marginRight: 15,
    width: '100%',
  },
}));

function LOPParts({ control }) {
  const classes = useCriteriaStyles();

  const dataSourceSummary = useDataSourceSummary('keywords');
  const [lopsAndPartsState, setLopsAndPartsState] = useRecoilState(
    lopsAndPartsStateAtom
  );

  const childComp = useMemo(() => {
    return (
      <Box className={classes.keyword} container elevation={10}>
        <Grid container elevation={10}>
          <Grid item className={classes.keyword_heading} xs={2}>
            <Typography className={classes.keyword_bolts}>LOP</Typography>
          </Grid>
          {/* <Grid item xs={4}> */}
          {lopsAndPartsState?.lopCriteria?.map((i, index) => {
            return (
              <React.Fragment key={index}>
                <Grid item xs={1}>
                  <Typography className={classes.date_value_label}>
                    {`${i.lop}`}
                  </Typography>
                </Grid>
                {index + 1 !== lopsAndPartsState?.lopCriteria?.length && (
                  <Divider orientation="vertical" variant="middle" flexItem />
                )}
              </React.Fragment>
            );
          })}
          {/* </Grid> */}

          <Grid item className={classes.keyword_heading} xs={2}>
            <Typography className={classes.keyword_bolts}>Fail Code</Typography>
          </Grid>
          {lopsAndPartsState?.lopCriteria?.map((i, index) => {
            return (
              <React.Fragment key={index}>
                <Grid item xs={1.5}>
                  <Typography className={classes.date_value_label}>
                    {`${i.failureCode}`}
                  </Typography>
                </Grid>
                {index + 1 !== lopsAndPartsState?.lopCriteria?.length && (
                  <Divider orientation="vertical" variant="middle" flexItem />
                )}
              </React.Fragment>
            );
          })}
        </Grid>
        <Divider />
        <Grid container elevation={10}>
          <Grid item className={classes.keyword_heading} xs={2}>
            <Typography className={classes.keyword_bolts}>Parts</Typography>
          </Grid>
          {lopsAndPartsState?.partsGroup?.parts?.map((i, index) => {
            return (
              <React.Fragment key={index}>
                <Grid item xs={1.5}>
                  <Typography className={classes.date_value_label}>
                    {i}
                  </Typography>
                </Grid>
                {index + 1 !== lopsAndPartsState?.partsGroup?.parts.length && (
                  <Divider orientation="vertical" variant="middle" flexItem />
                )}
              </React.Fragment>
            );
          })}
        </Grid>
      </Box>
    );
  }, []);

  return (
    <div>
      <Typography className={classes.heading}>LOP Parts</Typography>
      <DataSourceSummary dataSummary={dataSourceSummary} child={childComp} />
    </div>
  );
}
export default LOPParts;
