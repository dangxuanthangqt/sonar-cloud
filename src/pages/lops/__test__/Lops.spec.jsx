import React from 'react';
import { render, screen } from '../../../common/utils/testUtils';
import SalesCodes from '@/pages/sales-codes';
import LOPParts from '@/pages/summary/components/LOPParts';

describe('Data source', () => {
  it('should match snapshot', async () => {
    const { container } = render(<LOPParts />);
    expect(container).toMatchSnapshot();
  });

  it('should match element in document', async () => {
    const { getByText, getByRole } = render(<LOPParts />);

    expect(
      screen.getByRole('button', { name: 'Data sources' })
    ).toBeInTheDocument();

    expect(screen.getByText('Required')).toBeInTheDocument();

    expect(screen.getByText('Optional')).toBeInTheDocument();

    expect(screen.getByText('Not Applicable')).toBeInTheDocument();
  });
});
