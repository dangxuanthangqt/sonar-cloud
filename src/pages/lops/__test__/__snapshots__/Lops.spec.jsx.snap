// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data source should match snapshot 1`] = `
<div>
  <div>
    <p
      class="MuiTypography-root MuiTypography-body1 makeStyles-heading-5 css-16w2zvr-MuiTypography-root"
    >
      LOP Parts
    </p>
    <div
      class="makeStyles-container-14 MuiBox-root css-0"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1jni0wj-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-customIcon-15 css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandLessRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.29 8.71 6.7 13.3c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 10.83l3.88 3.88c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 8.71c-.38-.39-1.02-.39-1.41 0z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-8guo7j-MuiTypography-root"
        >
          Data sources
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-13z16su-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-dz5dyu-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-tfk0my-MuiTypography-root"
                >
                  Required
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-1196aik-MuiGrid-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-13z16su-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-dz5dyu-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-tfk0my-MuiTypography-root"
                >
                  Optional
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-1196aik-MuiGrid-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-container css-13z16su-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-dz5dyu-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-tfk0my-MuiTypography-root"
                >
                  Not Applicable
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-1196aik-MuiGrid-root"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-keyword-7 MuiBox-root css-0"
        elevation="10"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          elevation="10"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 makeStyles-keyword_heading-9 css-nwl7u4-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 makeStyles-keyword_bolts-8 css-16w2zvr-MuiTypography-root"
            >
              LOP
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 makeStyles-keyword_heading-9 css-nwl7u4-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 makeStyles-keyword_bolts-8 css-16w2zvr-MuiTypography-root"
            >
              Fail Code
            </p>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          elevation="10"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 makeStyles-keyword_heading-9 css-nwl7u4-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 makeStyles-keyword_bolts-8 css-16w2zvr-MuiTypography-root"
            >
              Parts
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
